add showcase page.

rabbihossain opened the pull request:
fixes #154

abuuzayr commented:
@rabbihossain thank you for this feature. it looks like theres a build failure. does it build fine on your machine?

rabbihossain commented:
@abuuzayr , yes it does build fine on my machine. I also cannot the see vercel build log here. Can you see the log and let me know why it's failing? 

You can also clone my branch and try running. I think that will work too.

abuuzayr commented:
Did you get this error @rabbihossain ?

![61401467-B755-4B2E-A43F-51CDA06C5366](https://user-images.githubusercontent.com/19371989/124351480-222dd480-dc2d-11eb-8bcb-31bba195791c.jpeg)

this isn’t from vercel tho, i have no access either. this is from a gitpod env https://aqua-fly-dnubrd2b.ws-us11.gitpod.io/

rabbihossain commented:
@abuuzayr  I made a typo while refactoring my code before my previous commit. Added e new commit. Vercel build is passing now.
My page can be seen on - https://blitzjs-com-7gzvuct8w-blitzjs.vercel.app/showcase

flybayer commented on app/core/components/ShowcaseThumbnail.js:
```diff
@@ -0,0 +1,10 @@
+export function ShowcaseThumbnail({title, thumbnail, URL, onClick}) {
+  return (
+    <div className="showcase-block" onClick={onClick}>
+      <img src={thumbnail} className="showcase-block-image" alt={title} />
```
Let's use our <Image> component as it has better perf

rabbihossain replied:
> Sweet, thank you!!
> 
> Personally I feel like it would be better for click to open the site instead of a modal. Especially because the modal is basically the same size as the thumbnail, so it doesn't give you much. What do you think?
> 
> Also, let's add a link to this page on the top and bottom menus

I added the modal because every website has a description and adding description text on main grid after each image, makes the UI very busy and inconsistent. 


> Let's use our  component as it has better perf

Are you suggesting Image component from "blitz"? Let me create a new commit then.

flybayer replied:
> I added the modal because every website has a description and adding description text on main grid after each image, makes the UI very busy and inconsistent.

Ok that makes sense. What about not adding a description? I'm not sure it adds that much value

> Are you suggesting Image component from "blitz"? Let me create a new commit then.

Yes

rabbihossain replied:
> Also, let's add a link to this page on the top and bottom menus
Can you give me a visual clue on that? Like what will be the link name and where it will sit with other menu items etc.

flybayer commented on app/pages/showcase.js:
```diff
@@ -0,0 +1,130 @@
+  )
+}
+
+const getStaticProps = async () => {
+  return {
+    props: {
+      showcase: showcaseList,
+    },
+    revalidate: 3 * 60 * 60, // 3 hours
+  }
+}
```
Since the list is hardcoded, we don't need static props. We can just use `showcaseList` in the component without it being a prop

rabbihossain replied:
Yeah, I agree too. Let me refactor this.

flybayer reviewed (commented):
Sweet, thank you!!

Personally I feel like it would be better for click to open the site instead of a modal. Especially because the modal is basically the same size as the thumbnail, so it doesn't give you much. What do you think?

Also, let's add a link to this page on the top and bottom menus

rabbihossain commented:
@flybayer I have refactored the code and removed the modal. Also added Image component from "blitz" instead of <img> tag. Can you check.

> Also, let's add a link to this page on the top and bottom menus

Can you give me a visual clue on that? Like what will be the link name and where it will sit with other menu items etc.

Once everything is done. let me know. Then I will rebase the all the commits and squash it into a single commit.

JuanM04 commented:
Awesome work! Also, don't forget to change the link at the bottom for something like "Open a PR!" and a link to this repo.

> Also, let's add a link to this page on the top and bottom menus
> Can you give me a visual clue on that? Like what will be the link name and where it will sit with other menu items etc.

It could be at the bottom, below **Community**. If it's added in the top navbar, you may need to change the responsive breakpoints.

> Once everything is done. let me know. Then I will rebase the all the commits and squash it into a single commit.

It isn't necessary, we do "Squash and merge" when merging the PR :smiley:

flybayer commented:
> Can you give me a visual clue on that? Like what will be the link name and where it will sit with other menu items etc.

@rabbihossain  In header, Add "Showcase" between Documentation and Releases. If needed we can shorten "Donate/Sponsor" to "Sponsor". And in footer add to bottom of "Community" link list.

rabbihossain commented:
@flybayer & @abuuzayr 
Sorry for a bit late. Was busy with some other works. Pushed a new commit & I think it should be good to merge now.

abuuzayr commented on app/pages/showcase.js:
```diff
@@ -0,0 +1,68 @@
+        <div className="font-secondary text-base text-gray-600 dark:text-gray-300">
+          Want to submit your website?{" "}
+          <a
+            href="https://github.com/blitz-js/blitzjs.com/"
```
```suggestion
            href="https://github.com/blitz-js/blitzjs.com/edit/main/app/core/utils/showcaseList.js"
```

JuanM04 reviewed (approved):
Amazing!

rabbihossain commented:
> Amazing!

Thanks :D

JuanM04 commented:
We've got a little problem. In the navbar, there is a new link to Flightcontrol, so maybe we need to relocate the showcase button

flybayer commented:
@rabbihossain I'm so sorry for the delay on my part. This is awesome, thank you so much!!
